File: garbage/solution/garbage-sub_4.py
class point:
    def __init__(self, x, w):
        self.x = x
        self.w = w

    def __lt__(self, obj):
        return self.x < obj.x

# read input
N, W, H = map(int, input().split())
pts = []
for i in range(N):
    x, y, w = map(int, input().split())
    pts.append(point(x, w))
pts.sort()

# two pointers
r = 0
result = 0
running_weight = 0
for l in range(N):
    left_border = pts[l].x
    right_border = left_border + W - 1
    while (r < N and pts[r].x <= right_border):
        running_weight += pts[r].w
        r += 1
    result = max(result, running_weight)
    running_weight -= pts[l].w
print(result)

File: garbage/submissions/partially_accepted/garbage-sub_4.cpp
#include <bits/stdc++.h>
using namespace std;

typedef long long ll;

int main() {
    // read input
    int N, W, H;
    cin >> N >> W >> H;
    vector<pair<int, int>> pts(N); // <x-position, weight>
    for (auto &e : pts) {
        int x, y, w;
        cin >> x >> y >> w;
        e = {x, w};
    }
    sort(pts.begin(), pts.end());

    // two pointers
    int r = 0;
    ll result = 0, running_weight = 0;
    for (int l = 0; l < N; l++) {
        int left_border = pts[l].first;
        int right_border = left_border + W - 1;
        while (r < N && pts[r].first <= right_border) {
            running_weight += pts[r].second;
            r++;
        }
        result = max(result, running_weight);
        running_weight -= pts[l].second;
    }
    cout << result << endl;
}
